hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/opt/OptUtils.java

```diff
@@ -148,6 +150,31 @@ public static HazelcastRelOptTable createRelTable(
         return new HazelcastRelOptTable(relTable);
     }
 
+    /**
+     * Get possible physical rels from the given subset.
+     * Every returned input is guaranteed to have a unique trait set.
+     *
+     * @param input Subset.
+     * @return Physical rels.
+     */
+    public static Collection<RelNode> extractPhysicalRelsFromAllSubsets(RelNode input) {
```

Comment:
Isn't below `extractPhysicalRelsFromSubset` enough?

Reply:
No, it's a different implementation.

Reply:
I replaced `extractPhysicalRelsFromAllSubsets` with `extractPhysicalRelsFromSubset`, tests seem to pass 🤔

Reply:
`extractPhysicalRelsFromSubset` wouldn't produce index scan, just normal scan.